ios/FluentUI/Vnext/PersonaGrid/PersonaGridItem.swift

```diff
@@ -0,0 +1,146 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+
+/// `MSFPersonaGridItemState` contains PersonaGridItem properties in addition to a subset of the MSFAvatarState protocol.
+///
+/// `onTapAction` provides tap gesture for PersonaGridItem.
+///
+@objc public protocol MSFPersonaGridItemState {
+    var image: UIImage? { get set }
+    var primaryText: String? { get set }
+    var secondaryText: String? { get set }
+    var size: MSFAvatarSize { get set }
+
+    var onTapAction: (() -> Void)? { get set }
+}
+
+/// Properties that make up PersonaGridItem content
+class MSFPersonaGridItemViewStateImpl: NSObject, ObservableObject, Identifiable, MSFPersonaGridItemState {
+    public var onTapAction: (() -> Void)?
+
+    @ObservedObject private var tokens: MSFPersonaGridItemTokens
+    private var avatarState: MSFAvatarState
+
+    var image: UIImage? {
+        get {
+            return avatarState.image
+        }
+
+        set {
+            avatarState.image = newValue
+        }
+    }
+
+    var primaryText: String? {
+        get {
+            return avatarState.primaryText
+        }
+
+        set {
+            avatarState.primaryText = newValue
+        }
+    }
+
+    var secondaryText: String? {
+        get {
+            return avatarState.secondaryText
+        }
+
+        set {
+            avatarState.secondaryText = newValue
+        }
+    }
+
+    var size: MSFAvatarSize {
+        get {
+            return avatarState.size
+        }
+
+        set {
+            avatarState.size = newValue
+        }
+    }
+
+    init(size: MSFPersonaGridSize, avatarState: MSFAvatarState) {
+        self.avatarState = avatarState
+        self.tokens = MSFPersonaGridItemTokens(size: size)
+
+        super.init()
+    }
+}
+
+public struct PersonaGridItem: View {
+    @ObservedObject var tokens: MSFPersonaGridItemTokens
+    @ObservedObject var state: MSFPersonaGridItemViewStateImpl
+    let avatarView: AvatarView
+    let gridSize: MSFPersonaGridSize
```

Comment:
This should be part of the state object as the client code could set a new size and have the view animating its size change.

Reply:
Agreed. I'd originally thought of `gridSize` being immutable at creation, but there's no reason it should have to be.